src/MongoDB.Driver.Core/Core/Bindings/CoreSession.cs

```diff
@@ -434,17 +435,41 @@ private void EnsureStartTransactionCanBeCalled()
         {
             if (_currentTransaction == null)
             {
-                return;
+                EnsureTransactionsAreSupported();
             }
+            else
+            {
+                switch (_currentTransaction.State)
+                {
+                    case CoreTransactionState.Aborted:
+                    case CoreTransactionState.Committed:
+                        break;
```

Comment:
Now that I'm commenting on this pull request I realize that when taking this code path we are not checking that transactions are supported, but I suppose it is reasonable that if they were supported when the current transaction was started that they should still be supported now?